kanidmd/src/lib/entry.rs

```diff
@@ -1755,24 +1765,20 @@ impl<VALID, STATE> Entry<VALID, STATE> {
     /// Assert if an attribute of this name is present, and one of it's values contains
     /// the following substring, if possible to perform the substring comparison.
     pub fn attribute_substring(&self, attr: &str, subvalue: &PartialValue) -> bool {
-        match self.attrs.get(attr) {
-            Some(v_list) => v_list
-                .iter()
-                .fold(false, |acc, v| if acc { acc } else { v.contains(subvalue) }),
-            None => false,
-        }
+        self.attrs
+            .get(attr)
+            .map(|vset| vset.substring(subvalue))
+            .unwrap_or(false)
```

Comment:
How come `self.attrs.get(attr)` used to return an iterable while it now returns a `PartialValue` (or `Value`, I can't infer here)? What were we iterating through before?

Reply:
Because the former code was really really old before I knew how to use map nicely. So the some(vlist) now changes to .map(|vset| and then we just call valueset substring's impl which has it's own internal iterator instead.